### ISSUE TITLE
Wrong use of "local" in methods of Public

### ISSUE THREAD

**Issue by bocops** [andregasser/bigbone#75]
[Public.getPublic()](https://github.com/andregasser/bigbone/blob/e2c2b5e79708139a56c97af1e07c77502acc1da0/bigbone/src/main/kotlin/social/bigbone/api/method/Public.kt#L63) and [Public.getTag()](https://github.com/andregasser/bigbone/blob/e2c2b5e79708139a56c97af1e07c77502acc1da0/bigbone/src/main/kotlin/social/bigbone/api/method/Public.kt#L88) have a `local` parameter that is used by `getLocal...` and `getFederated...` functions in the same file.

The `local` parameter is only appended if `true` - if `false`, nothing happens. I believe that a `remote` parameter should be appended in these cases. Compare [https://docs.joinmastodon.org/methods/timelines/#public](https://docs.joinmastodon.org/methods/timelines/#public)

**Comment by andregasser** [comment]
I ran some tests against the `https://fosstodon.org/api/v1/timelines/public` and here are the results:

| local | remote | Response |
|---|---|---|
| false | false | returns statuses from Fosstodon but also statuses posted on other servers (local statuses and remote statuses combined) | 
| true | false | returns only statuses posted on Fosstodon (local statuses) | 
| false | true | returns only remote statuses from other Mastodon servers than Fosstodon (remote statuses) |
| true | true | returns an empty array | 

According to the docs, `local` and `remote` have a default value of `false`. So if we just switch `local` from `true` to `false` and do not set the remote flag, we cover the first two rows in the table only, the third is missing, means, if someone using our library wants to get only remote statuses, he could not do that currently, as we do not pass the remote flag to the Mastodon endpoints.

My proposal would be to also set the remote flag and make sure we can cover all three cases (rows 1 to 3 in the table).

**Comment by bocops** [comment]
That means we should either add a single query parameter (`local=true` or `remote=true`) to get the local or federated timelines, or neither, in which case we get the combined timeline. We should never use both parameters at the same time.

That use of boolean values is making my head hurt, but I can still implement it. ;)

**Comment by andregasser** [comment]
The Mastodon API uses these two flags to understand what kind of statuses the library user wants to retrieve. It's either local statuses only, remote statuses only or local and remote statuses combined. 

Since we're anyways introducing breaking changes with the 2.0.0 release, what about simplifying the method signatures for the library user? Currently, we have two methods for fetching the public timeline. With these signatures, the library user cannot fetch only remote statuses, so we do not cover all possible cases (as outlined in the table above):

```kotlin
@JvmOverloads
fun getLocalPublic(range: Range = Range()) = getPublic(true, range)

@JvmOverloads
fun getFederatedPublic(range: Range = Range()) = getPublic(false, range)
```

What about changing the code into something like this:

```kotlin
enum class StatusOrigin {
    LOCAL, REMOTE, LOCAL_AND_REMOTE
}

@JvmOverloads
fun getPublic(range: Range = Range(), origin: StatusOrigin = StatusOrigin.LOCAL_AND_REMOTE) { 
    ...
}
```

In method `getPublic` we could then set the flags `local` and `remote` according to the table in order to get the desired data. I think that would simplify the API for the library user. This way, we would also cover all three cases.

**Comment by bocops** [comment]
Yes, that's the best way going forward. If we do that, I think we should just remove the overloaded functions `getLocal...` and `getRemote...`.

Do you happen to know if `local`/`remote` is used anywhere else? If it is, there might be a better place for the suggested enum than in Public.kt directly.
